### PR TITLE
Structures globals within plan to lie within catalogs

### PR DESCRIPTION
**Pull request by johnedquinn**
## Relevant Issues
- None

## Description
- Structures globals within plan to lie within catalogs

## Other Information
- Updated Unreleased Section in CHANGELOG: **YES**

- Any backward-incompatible changes? **YES**

- Any new external dependencies? **NO**

- Do your changes comply with the [Contributing Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CONTRIBUTING.md)
  and [Code Style Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CODE_STYLE.md)? **YES**

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-plan/src/main/resources/partiql_plan.ion [hunk 1]
```diff
@@ -8,15 +8,22 @@ imports::{
...
+// - Eventually, TODO functions may be resolved to a specific namespace within a catalog.
+catalog::{
+  name: string,
+  values: list::[value],
+  _: [
+    value::{
+      path: list::[string], // This should NOT be empty.
+      type: static_type
+    }
+  ]
```
Comment (RCHowell) [inline partiql-plan/src/main/resources/partiql_plan.ion:26]:
```suggestion
catalog::{
  name: string,
  symbols: list::[symbol],
}
  
symbol::[
    value::{
        path: list::[string], // This should NOT be empty.
        type: static_type
    }
    // TODO fn
    _::[
      ref::{
         catalog: int,
         symbol: int,
      }
    ]
]
```



This structure also aligns with https://en.wikipedia.org/wiki/Symbol_table

> The minimum information contained in a symbol table used by a translator and [intermediate representation](https://en.wikipedia.org/wiki/Intermediate_representation) (IR) includes the symbol's name and its location or address.

File: partiql-plan/src/main/resources/partiql_plan.ion [hunk 2]
```diff
@@ -70,7 +77,8 @@ rex::{
     },
 
     global::{
-      ref: int,
+      catalog_ref: int,  // The index of the relevant catalog in the plan's catalogs list
+      value_ref: Int     // The index of the relevant value in a catalog's values list
```
Comment (RCHowell) [inline partiql-plan/src/main/resources/partiql_plan.ion:81]:
```suggestion
    ref: '.symbol.ref'
```

**Review by RCHowell — CHANGES_REQUESTED** [review]

File: partiql-plan/src/main/resources/partiql_plan.ion (hunk 1 above)
Comment (johnedquinn) [reply partiql-plan/src/main/resources/partiql_plan.ion]: I like the use of `ref` a lot. Modified slightly for the nesting, but just pushed.

**Review by RCHowell — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -48,6 +48,7 @@ Thank you to all who have contributed!
   - The new plan is fully resolved and typed.
   - Operators will be converted to function call. 
 - Changes the return type of `filter_distinct` to a list if input collection is list
+- **BREAKING**: Globals within the logical plan are now represented in a structured manner to reflect the existence of catalogs. Each value is now mandated to be case-sensitive (we have moved from Identifiers to string literals).
```
Comment (RCHowell) [inline CHANGELOG.md:51]: These were unreleased AFAIK

**Review by RCHowell — CHANGES_REQUESTED** [review]

**Review by RCHowell — APPROVED** [review]

**Pull request merged**